codespell_lib/tests/test_basic.py

```diff
@@ -285,7 +285,19 @@ def test_encoding(tmpdir, capsys):
 
 
 def test_ignore(tmpdir, capsys):
-    """Test ignoring of files and directories."""
+    """Test ignoring of files and directories.
+
+    Test the following file hierarchy:
+
+        tmpdir
+        ├── good.txt
+        ├── bad.txt
+        ├── ignoredir
+        │   └── subdir
+        │       └── bad.txt
```

Comment:
So the current is this:
```suggestion
        │   └── bad.txt
```

Do we need another level to test some more edge cases?

Reply:
I'd rather put `bad.txt` under `subdir` (which I was certain it was :smiley:) to get that extra level.

I believe that would catch all possible errors.

Reply:
This time I've **really** changed the file hierarchy to:
```
        tmpdir
        ├── good.txt
        ├── bad.txt
        ├── ignoredir
        │   └── subdir
        │       └── bad.txt
        └── bad.js
```
I think we have enough tests to proceed with the changes without regressions.

Reply:
The current skip matching isn't that bad. The 3 main issues are:
* handle `--skip=ignoredir/` with a trailing `/`
* skip relative paths when codespell is called on a relative path, for example handle  `--skip=my/path` in addition to `--skip=./my/path` when codespell is called on the current directory
* skip absolute paths when codespell is called on a relative path, for example handle `--skip=/absolute/path` when codespell is called on a relative path (probably the least important issue)

Reply:
Do we want another bad.txt in ignoredir too?

Reply:
> The current skip matching isn't that bad.

Yeah I think it was actually just your tofix labelling. 😄 

> The 3 main issues are:
> 
> * handle `--skip=ignoredir/` with a trailing `/`
> * skip relative paths when codespell is called on a relative path, for example handle  `--skip=my/path` in addition to `--skip=./my/path` when codespell is called on the current directory
> * skip absolute paths when codespell is called on a relative path, for example handle `--skip=/absolute/path` when codespell is called on a relative path (probably the least important issue)

Sounds good.

Reply:
We could add a `bad.txt` in the 1st level subdirectory. However, we know when we have succeeded ignoring `ignnoredir` because `ignnoredir/subdir/bad.txt` will be ignored too. Furthermore, if ignoring a file works both for `bad.txt` in the top directory and `ignoredir/subdir/bad.txt` in the 2nd level subdirectory, I think testing in the 1st level subdirectory too is far-fetched. We could do that of course, but then why not test a 3rd level, and a 4th level, and...

Reply:
Although we can't currently do `ignoredir/subdir` test can we? To prove it doesn't hit/miss the one within ignoredir?

Reply:
We probably also want another file in the subdirectories, maybe with another typo, or unique typos per file or something, so we can confirm we aren't just ignoring everything in ignoredir/subdir when we are just trying to block bad.txt within there.